lib/systems/default.nix

```diff
@@ -91,7 +95,26 @@ let
         && final.parsed.kernel == platform.parsed.kernel;
       isCompatible = _: throw "2022-05-23: isCompatible has been removed in favor of canExecute, refer to the 22.11 changelog for details";
       # Derived meta-data
-      useLLVM = final.isFreeBSD || final.isOpenBSD;
+
+      toolchain =
+        /**/ if final.isDarwin then "apple"
+        else if final.isFreeBSD || final.isOpenBSD then "llvm"
+        else "gnu";
+
+      cc = if final.toolchain == "apple" || final.toolchain == "llvm" then
+        "clang"
+      else "gcc";
+
+      bintools = if final.toolchain == "llvm" then
+        "llvm"
+      else if final.toolchain == "apple" then
+        "apple"
+      else "gnu";
+
+      libcxx =
+        /**/ if final.toolchain == "llvm" then "libcxx"
+        else if final.toolchain == "gnu" then "libstdcxx"
+        else null;
```

Comment:
Are `hostPlatform.libcxx == "libcxx"` and `stdenv.cc.libcxx != null` ever going to disagree? If not, we can probably drop this?

If we keep it, we should probably give it a name that doesn’t overlap with one of the options, to avoid confusion. There’s also the distinction of `libc++abi` vs. `libcxxrt` even when using `libc++`.

Reply:
> Are `hostPlatform.libcxx == "libcxx"` and `stdenv.cc.libcxx != null` ever going to disagree?

Yes.

```
nix-repl> legacyPackages.aarch64-linux.stdenv.cc.libcxx
null

nix-repl> legacyPackages.aarch64-linux.stdenv.hostPlatform.libcxx
"libstdcxx"
```

`stdenv.cc.libcxx` doesn't look like a reliable way to get the C++ stdlib.

> If we keep it, we should probably give it a name that doesn’t overlap with one of the options, to avoid confusion. There’s also the distinction of `libc++abi` vs. `libcxxrt` even when using `libc++`.

So what would be a better name? `cxx-std`?

Reply:
I’m not too fussed about the name; `cxxlib` seems consistent with the other choices in this PR. (That result seems consistent to me, though? `aarch64-linux` doesn’t use `libc++`, so `stdenv.cc.libcxx` is `null`.)